lib/web/userpreferences.go

```diff
@@ -181,10 +181,17 @@ func userPreferencesResponse(resp *userpreferencesv1.UserPreferences) *UserPrefe
 	return jsonResp
 }
 
-func clusterPreferencesResponse(resp *userpreferencesv1.ClusterUserPreferences) ClusterUserPreferencesResponse {
-	return ClusterUserPreferencesResponse{
-		PinnedResources: resp.PinnedResources.ResourceIds,
+func clusterPreferencesResponse(prefs *userpreferencesv1.ClusterUserPreferences) ClusterUserPreferencesResponse {
+	resp := ClusterUserPreferencesResponse{
+		PinnedResources: make([]string, 0),
```

Comment:
This is not necessary. The `append` below will work just fine on a nil slice.

```suggestion
```

Reply:
We _want_ it to be an empty array instead of `null`, however, I think we have some checks on the webui anyway along the lines of `if falsy return []`, so I can remove this. This is good info, thanks!